### PR TITLE
Refactor some usage of shell into native kubernetes modules + small fixes

### PR DESCRIPTION
**Pull request by lentzi90**
The major part of this PR is converting use of the Ansible `shell` module into `k8s_info`. In addition to this it changes all `include`s to `include_tasks` to make the behavior more predictable and avoid problems later when it is deprecated. (See [the documentation](https://docs.ansible.com/ansible/2.9/modules/include_module.html#include-module).)

### CODE REVIEW

File: vm-setup/roles/v1aX_integration_test/tasks/verify.yml
Comment (macaptain) [inline vm-setup/roles/v1aX_integration_test/tasks/verify.yml:110]: `grep -v` is an inverted match, so in the shell we are listing the pods that are _not_ in running phase and waiting until the output is empty. However, I think the field selector with the until clause here is doing the opposite? If so, I expect we have to fix this.

These "check" task names could be clearer as well to indicate that we're waiting for a condition.

**Review by macaptain — CHANGES_REQUESTED** [review]
I like the extra information we will get in the Ansible output. We need to check that the conditions are exactly the same as the old shell though.

File: vm-setup/roles/v1aX_integration_test/tasks/verify.yml
Comment (lentzi90) [reply vm-setup/roles/v1aX_integration_test/tasks/verify.yml:110]: Oh gotcha! :facepalm:

**Review by macaptain — APPROVED** [review]
This looks good to me, and I like this change. In particular, the increased log output could be helpful for debugging problems in future. I don't have permissions to add approvals which will allow merging, so other members of the project will have to take a look.

File: 01_prepare_host.sh
Comment (lentzi90) [inline 01_prepare_host.sh:123]: Note the missing `/` before `home`. This creates a folder on the relative path which doesn't seem to be used for anything. Later references to `.cluster-api` all use the absolute path `/home/${USER}/.cluster-api`.

Comment (kashifest) [reply 01_prepare_host.sh:123]: @furkatgofurov7 any idea why it was so and should we add a `/` before `home` instead of removing it?

File: vm-setup/roles/v1aX_integration_test/tasks/cleanup.yml
Comment (kashifest) [inline vm-setup/roles/v1aX_integration_test/tasks/cleanup.yml:27]: I think we also need to parameterize this since we would have two versions of these soon

File: vm-setup/roles/v1aX_integration_test/tasks/cleanup.yml
Comment (kashifest) [inline vm-setup/roles/v1aX_integration_test/tasks/cleanup.yml:52]: I think we also need to parameterize this since we would have two versions of these soon

File: vm-setup/roles/v1aX_integration_test/tasks/cleanup.yml
Comment (kashifest) [inline vm-setup/roles/v1aX_integration_test/tasks/cleanup.yml:63]: same

File: vm-setup/roles/v1aX_integration_test/tasks/cleanup.yml
Comment (kashifest) [inline vm-setup/roles/v1aX_integration_test/tasks/cleanup.yml:74]: same

**Review by kashifest — COMMENTED** [review]
Lgtm. Some comments inline

File: 01_prepare_host.sh
Comment (furkatgofurov7) [reply 01_prepare_host.sh:123]: Please see the related [issue](https://github.com/metal3-io/metal3-dev-env/issues/470) and [fix](https://github.com/metal3-io/metal3-dev-env/pull/485) where these were added. IMO we need `.cluster-api ` part with missing `/` but `version.yaml` most probably might not be needed.

File: vm-setup/roles/v1aX_integration_test/tasks/cleanup.yml
Comment (lentzi90) [reply vm-setup/roles/v1aX_integration_test/tasks/cleanup.yml]: Done

File: vm-setup/roles/v1aX_integration_test/tasks/cleanup.yml
Comment (lentzi90) [reply vm-setup/roles/v1aX_integration_test/tasks/cleanup.yml]: Done

File: vm-setup/roles/v1aX_integration_test/tasks/cleanup.yml
Comment (lentzi90) [reply vm-setup/roles/v1aX_integration_test/tasks/cleanup.yml]: Done

File: vm-setup/roles/v1aX_integration_test/tasks/cleanup.yml
Comment (lentzi90) [reply vm-setup/roles/v1aX_integration_test/tasks/cleanup.yml]: Done

**Comment by lentzi90** [comment]
Thanks for the comments @kashifest !
I used the `CAPI_VERSION` and `CAPM3_VERSION` that were already defined to parametrize the API versions. Do you think we should also parametrize the BMO version? I didn't find a variable for this.

File: 01_prepare_host.sh
Comment (lentzi90) [reply 01_prepare_host.sh:123]: That's weird. I haven't had any problems without this `.cluster-api` folder. :thinking:  Could it have been some strange value set for `$HOME`? I mean `clusterctl` should use `$HOME/.cluster-api`, right? Unless we set `--config` to something different.

Comment (lentzi90) [reply 01_prepare_host.sh:123]: Could it be actually this issue https://github.com/kubernetes-sigs/cluster-api/issues/3573 that was the root cause for #470? The timing is suspicious and the error seems to be very much the same.

**Review by kashifest — APPROVED** [review]

**Comment by kashifest** [comment]
/hold
Squash the commits please

**Comment by lentzi90** [comment]
Looks like I need to check also for empty list, not just unreachable API.
```
TASK [v1aX_integration_test : Wait until cluster becomes provisioned.] *********
task path: /home/****/tested_repo/vm-setup/roles/v1aX_integration_test/tasks/verify.yml:6
fatal: [localhost]: FAILED! => {
    "msg": "The conditional check '(provisioned_cluster is succeeded) and (provisioned_cluster.resources[0].status.phase | lower == \"provisioned\")' failed. The error was: error while evaluating conditional ((provisioned_cluster is succeeded) and (provisioned_cluster.resources[0].status.phase | lower == \"provisioned\")): list object has no element 0"
}
```

Will try to address this. It seems very rare though since I haven't seen it in any of the previous runs.

File: 01_prepare_host.sh
Comment (furkatgofurov7) [reply 01_prepare_host.sh:122]: Strange, how come then we were facing that reported issue? Most probably CAPI had issue and we fixed it before them and now the fix seems to be redundant.

**Comment by lentzi90** [comment]
I'm confused about the version changes. Maybe someone can explain?
After #685 was merged I get an error in the integration tests:
```
Failed to find API for resource with apiVersion \"cluster.x-k8s.io/v1alpha4\" and kind \"Cluster\"
```
So CAPI_VERSION must be set to v1alpha4 that is why it is trying to check for it. But when I check the CRD there is only v1alpha2 and v1alpha3 there.

I noticed also that this is hard-coded in the templates, e.g. here: https://github.com/metal3-io/metal3-dev-env/blob/8b61eaee442c7468d07f0209ed2e43f54ef16017/vm-setup/roles/v1aX_integration_test/templates/cluster-template-cluster.yaml#L1
I'm not sure really how these things should work but I think `CAPI_VERSION` should be set to `v1alpha3` in the tests until the CRDs support `v1alpha4`.

**Comment by lentzi90** [comment]
Based on offline discussion we are going with hard coding the CAPI version until v1alpha4 support has landed. Then we should do a followup and parameterize it.
/test-integration
/test-v1a4-centos-integration

**Pull request merged**
